### PR TITLE
golangci checks in github action, and fixes initial linter issues

### PR DESCRIPTION
**Pull request by another-rex**
- Uses a modified .golangci config file from osv-detector. Some harder lints disabled for now.
- Fixes some of the linting errors.
- Add golangci to be an optional prereq in CONTRIBUTING.md
- Adds golangci github action

### CODE REVIEW

**Review by G-Rath — APPROVED** [review]
looks good to me! fwiw the only one I think we disagree on is `nlreturn` - I personally find it makes for more readable code, but I can live with it not being enforced if you feel differently 🤷

**Comment by another-rex** [comment]
I don't feel too strongly about it either way. I added it back in with a block-size of 2 so it doesn't enforce cases where we have 
```
if err != nil {
    print_error(err)
    return err
}
```

File: internal/sbom/spdx.go [hunk 1]
```diff
@@ -10,10 +12,10 @@ import (
 )
 
 type SPDX struct{}
-type spdxLoader func(io.Reader) (*spdx.Document2_2, error)
+type SPDXLoader func(io.Reader) (*spdx.Document2_2, error)
```
Comment (oliverchang) [inline internal/sbom/spdx.go:15]: is this strictly needed? This makes the type public.

I think the rule is that the casing is consistent for acronyms (i.e. either always UPPERCASE or lowercase).

**Review by oliverchang — APPROVED** [review]
LGTM with minor comments. thanks!

File: run_lints.sh [hunk 2]
```diff
@@ -2,4 +2,4 @@
 
 set -ex
 
-go vet ./...
+golangci-lint run
```
Comment (G-Rath) [inline run_lints.sh:5]: This is assuming that `golangci-lint` is installed and that is the correct version - my understanding is that you can use `go run` here instead to help handle both of those:

```suggestion
go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.50.1 run ./... --max-same-issues 0
```

(`--max-same-issues` is what I've got over in `osv-detector` as I feel it's better to have a lint tool show you all the results at once - feel free to remove if you disagree)

File: internal/sbom/spdx.go (hunk 1 above)
Comment (another-rex) [reply internal/sbom/spdx.go]: Reverted this change

File: run_lints.sh (hunk 2 above)
Comment (another-rex) [reply run_lints.sh]: Thanks, that makes sense.

**Pull request merged**